algebird-core/src/main/scala/com/twitter/algebird/SetDiff.scala

```diff
@@ -0,0 +1,89 @@
+/*
+Copyright 2016 Twitter, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+ */
+
+package com.twitter.algebird
+
+/**
+ * `SetDiff` is a class that represents changes applied to a set. It
+ * is in fact a Set[T] => Set[T], but doesn't extend Function1 since
+ * that brings in a pack of methods that we don't necessarily want.
+ */
+sealed abstract case class SetDiff[T] private (add: Set[T], remove: Set[T]) { self =>
```

Comment:
Hey! Thanks for the review - I was cueing off of the approach here: https://gist.github.com/tpolecat/a5cb0dc9adeacc93f846835ed21c92d2

Can you explain what you mean by awkward?